pkg/grpc/server.go

```diff
@@ -220,7 +235,11 @@ func (a *apiImpl) listenOnLocalEndpoint(server *grpc.Server) pipeconn.DialContex
 		if err := server.Serve(lis); err != nil {
 			log.Fatal(err)
 		}
-		log.Fatal("The local API server should never terminate")
+
+		// If finishing with no error and shutdown was not requested, crash
+		if !a.grpcShutdownRequested.IsDone() {
+			log.Fatal("The local API server should never terminate unless explicitly requested")
```

Comment:
I don't understand this code, it is contradictory to me. 
First I understand "A API server should only terminate if request", and than crashes? 
I would expect a warning or error log and continue the program execution.

Reply:
`a.grpcShutdownRequested.IsDone()` indicates that `.Stop()` was called on the grpc server instance. 

If the listeners stopped, but the conditional `!a.grpcShutdownRequested.IsDone()` (negated) is `true` it means that the `server.Serve` function returned an error without `.Stop()` being called.

In this case, we want to keep the same behavior we had before: raise a panic in central or sensor.

If we don't crash here there might be a problem that something went wrong in the gRPC server; it fully stopped; but the application is still running. IMO it's preferable to keep the current behavior of crashing rather than running without a gRPC server.

Reply:
Ah, understood, this makes sense.

Suggestion make the comment more clear:

 - Remove the comment above the `if` statement, it is redundant with the log message
 - Log message suggestion:

```suggestion
			log.Fatal("Unexpected local API server termination.")
```